Framework/DataObjects/inc/MantidDataObjects/VectorColumn.h

```diff
@@ -122,6 +123,59 @@ template <class Type> class DLLExport VectorColumn : public API::Column {
     throw std::runtime_error("VectorColumn is not assignable from double.");
   }
 
+  /// Reference to the data.
+  const std::vector<std::vector<Type>> &data() const { return m_data; }
+
+  bool equals(Column *otherColumn, double tolerance) const override {
+    auto convertedOtherColumn = dynamic_cast<VectorColumn *>(otherColumn);
+    if (!convertedOtherColumn) {
+      return false;
+    }
+    if (convertedOtherColumn->size() != this->size() ||
+        convertedOtherColumn->type() != this->type()) {
+      return false;
+    }
+    auto otherData = convertedOtherColumn->data();
+    for (size_t i = 0; i < m_data.size(); i++) {
+      if (m_data[i].size() != otherData[i].size()) {
+        return false;
+      }
+      for (size_t j = 0; j < m_data[i].size(); j++) {
+        if (abs(m_data[i][j] - otherData[i][j]) > tolerance) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  bool equalsRelErr(Column *otherColumn, double tolerance) const override {
+    auto convertedOtherColumn = dynamic_cast<VectorColumn *>(otherColumn);
+    if (!convertedOtherColumn) {
+      return false;
+    }
+    if (convertedOtherColumn->size() != this->size() ||
+        convertedOtherColumn->type() != this->type()) {
+      return false;
+    }
+    auto otherData = convertedOtherColumn->data();
```

Comment:
`const auto &` here too.

Reply:
can you tell that I've been mostly in python for a while?

Reply:
I usually end up writing `raise` everywhere :)

Reply:
I keep forgetting these ;